src/app/credExplorer/pagerankTable/aggregation.test.js

```diff
@@ -1,7 +1,422 @@
+        for (const connection of aggregation.connections) {
+          expect(
+            NodeAddress.hasPrefix(
+              connection.source,
+              aggregation.nodeType.prefix
+            )
+          ).toBe(true);
+        }
+      }
+    });
+    it("when multiple node types match, it uses the most specific one", () => {
+      const {nodeTypes, scoredConnections} = example();
+      const aggregations = aggregateByNodeType(
+        [scoredConnections.rfz4],
+        [nodeTypes.empty, nodeTypes.zap]
+      );
+      expect(aggregations).toHaveLength(1);
+      // It used the zap nodeType rather than the empty nodeType b/c it was more specific
+      expect(aggregations[0].nodeType).toEqual(nodeTypes.zap);
+    });
+    it("summarizes the group size appropriately", () => {
+      const {nodeTypesArray, scoredConnectionsArray} = example();
+      const aggregations = aggregateByNodeType(
+        scoredConnectionsArray,
+        nodeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        expect(aggregation.summary.size).toEqual(
+          aggregation.connections.length
+        );
+      }
+    });
+    it("summarizes the group score appropriately", () => {
+      const {nodeTypesArray, scoredConnectionsArray} = example();
+      const aggregations = aggregateByNodeType(
+        scoredConnectionsArray,
+        nodeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        const totalScore = aggregation.connections.reduce(
+          (val, connection) => val + connection.connectionScore,
+          0
+        );
+        expect(aggregation.summary.score).toEqual(totalScore);
+      }
+    });
+    it("does not create empty groups", () => {
+      const {nodeTypesArray, scoredConnectionsArray} = example();
+      const aggregations = aggregateByNodeType(
+        scoredConnectionsArray,
+        nodeTypesArray
+      );
+      for (const aggregation of aggregations) {
+        expect(aggregation.summary.size).not.toEqual(0);
+      }
+    });
+    it("handles the case where there are no connections", () => {
+      const {nodeTypesArray} = example();
+      const aggregations = aggregateByNodeType([], nodeTypesArray);
+      expect(aggregations).toHaveLength(0);
+    });
+    it("errors if any connection has no matching type", () => {
+      const {scoredConnectionsArray} = example();
+      const shouldFail = () => aggregateByNodeType(scoredConnectionsArray, []);
+      expect(shouldFail).toThrowError("No matching NodeType");
+    });
+    it("sorts the aggregations by total score", () => {
+      const {nodeTypesArray, scoredConnectionsArray} = example();
+      const aggregations = aggregateByNodeType(
+        scoredConnectionsArray,
+        nodeTypesArray
+      );
+      let lastSeenScore = Infinity;
+      for (const aggregation of aggregations) {
+        const score = aggregation.summary.score;
+        expect(lastSeenScore >= score).toBe(true);
```

Comment:
Prefer `expect(lastSeenScore).toBeGreaterThanOrEqual(score)`, yes?

Reply:
Sure, done (here and elsewhere)